src/figuretype/animal.c

```diff
@@ -142,10 +142,7 @@ void figure_sheep_action(figure *f)
             if (f->wait_ticks > 400) {
                 f->wait_ticks = f->id & 0x1f;
                 f->action_state = FIGURE_ACTION_197_HERD_ANIMAL_MOVING;
-                f->destination_x = m->destination_x
-                    + formation_layout_position_x(FORMATION_HERD, f->index_in_formation);
-                f->destination_y = m->destination_y
-                    + formation_layout_position_y(FORMATION_HERD, f->index_in_formation);
+                formation_layout_position(FORMATION_HERD, f->index_in_formation, m, &f->destination_x, &f->destination_y);
```

Comment:
You could just call `map_grid_bound` from here, no?

No need for a whole new function with header includes.

Reply:
@crudelios i didn’t end up changing the type as I didn’t want to muck with the serialization.

Reply:
Fair enough 😉

Reply:
Finally got a chance to look at the earthquake tests. Unsurprisingly, the issue is the destination_x/y for sheep. The new save has a (now bounded) value of 0 and the old destination_x/y is 255.

Reply:
I've gone ahead and added an exception to the test code but let me know if y'all want to go with something else.

Reply:
The exception looks good to me, let's go ahead with this.